Kleisli.md

```diff
@@ -0,0 +1,169 @@
+autoscale: true
+footer: ![Arrow](arrow-brand-128x128.png) [@47deg](https://twitter.com/47deg) :: [Λrrow](http://arrow-kt.io) :: [http://arrow-kt.io/docs/datatypes/kleisli/](http://arrow-kt.io/docs/datatypes/kleisli/)
+slidenumbers: true
+
+# Kleisli 
+
+__`Kleisli`__ is a data type used in __Λrrow__ to model a transformation 
+
+from one type into another inside a monadic context
+
+---
+
+# Kleisli :: ADT
+
+__`Kleisli`__ represents an arrow from __`<D>`__ to a monadic value __`Kind<F, A>`__.
+
+That means when we create a `Kleisli<Id,Int,Double>` 
+
+we are doing `<Int> -> Id<Double>`
+
+---
+
+# Kleisli :: ADT
+
+Inside the `Kleisli` we specify the transformation.
+
+If we want to transform from the `Int` to the `Id<Double>`
+
+```kotlin
+val doubleIdKleisli = Kleisli { number: Int ->
+  Id.pure(number.toDouble())
+}
+
+val doubleId = doubleIdKleisli.run(1)
+//Id(1.0)
+```
+
+---
+
+# Kleisli :: Local
+
+The __`local`__ function allows doing a conversion inside the `Kleisli` 
+
+to the original input value before the Kleisli will be executed,
+
+creating a Kleisli with the input type of the conversion
+
+```kotlin
+val localId = doubleIdKleisli.local { optNumber :Option<Int> -> 
+optNumber.getOrElse { 0 } 
+}.run(None)
+//Id(0.0)
+```
+
+---
+
+# Kleisli :: Ap
```

Comment:
Instead of `Ap` we should feature `ask` which is a more popular combinator in Kleisli:
```
...binding {
  val (n, d) = Kleisli.ask<ForOption, Config>().bind()
  //extracts a value from the environment
}
```